cloudify_hostpool/config/config.py

```diff
@@ -0,0 +1,126 @@
+    if default is not None:
+        default_auth = default.get('auth')
+    _add_servers(storage, config.get('hosts'))
+
+
+def _ip2long(ip):
+    # throws socket.error if IP is not valid
+    return struct.unpack('!L', socket.inet_aton(ip))[0]
+
+
+def _long2ip(num):
+    return socket.inet_ntoa(struct.pack('!L', num))
+
+
+def _get_ibitmask(mask):
+    return (2L << (32-int(mask))-1) - 1
+
+
+def _get_broadcast_long(subnet, mask):
+    bin_sub = _ip2long(subnet)
+    bin_imask = _get_ibitmask(int(mask))
+    return bin_sub | bin_imask
+
+
+def _get_subnet_and_mask(ip_range):
+    # throws socket.error if IP is not valid
+    regex = re.compile("\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}/\d{1,2}")
+    result = regex.findall(ip_range)
+    if len(result) != 1:
+        raise socket.error("illegal IP address string")
+    s, m = ip_range.split('/')
+    return s, m
+
+
+def _get_subnet_hosts(subnet, mask):
+    """ Subnet hosts generator.
+    Yields each server address given in ip range.
+    """
+    bin_sub = _ip2long(subnet)
+    bin_broadcast = _get_broadcast_long(subnet, mask)
+    for address in range(bin_sub + 1, bin_broadcast):
+        yield _long2ip(address)
+
+
+def _add_servers(storage, servers):
+    """ Add server to database creating the server structure (dictionary)
+        server = {
+            'public_ip': ip address or hostname,
+            'private_ip': ip address or hostname,
+            'port': port to communicate to,
+            'auth': a dictionary with 'username' and 'keyfile' or 'password',
+            'alive': flag that will inform if this server has been reachable
+            recently,
+            'reserved': flag informing if this server is to be assigned at the
+            moment
+        }
+    """
+    for server in servers:
+        if server.get('private_ip') is not None:
+            server['alive'] = False
+            server['reserved'] = False
+            if server.get('auth') is None:
+                server['auth'] = dict(default_auth)
+            try:
+                server['port'] = server['auth'].pop('port')
+            except KeyError:
+                raise ConfigError("No authorisation port given")
+            storage.add_server(server)
+        elif server.get('ip_range') is not None:
+            subnet, mask = _get_subnet_and_mask(server.get('ip_range'))
+            servers_list_gen = _get_subnet_hosts(subnet, mask)
+            for server_ip in servers_list_gen:
+                if server.get('auth') is not None:
+                    auth = dict(server['auth'])
+                elif default_auth:
+                    auth = dict(default_auth)
+                else:
+                    raise ConfigError("No authorisation given")
+                try:
+                    s = dict(private_ip=server_ip,
+                             auth=auth,
+                             alive=False,
+                             reserved=False,
+                             port=auth.pop('port'))
```

Comment:
We were thinking about opimization for sqlite searches but you're right - this could be handled by storage part. This means I have to change also sqlite part.